static/js/util/profile_edit.js

```diff
@@ -12,72 +15,141 @@ import moment from 'moment';
 
 // bind to this.boundTextField in the constructor of a form component
 // to update text fields when editing the profile
-export function boundTextField(key, label, error) {
-  const { updateProfile, profile } = this.props;
+// we pass in a keyset looking like this:
+//
+// ["top-level-key", index, "nested_object_key"] or just ["top_level_key"]
+export function boundTextField(keySet, label) {
+  const {
+    profile,
+    errors,
+    updateProfile
+  } = this.props;
   let onChange = e => {
-    let clone = Object.assign({}, profile);
-    clone[key] = e.target.value;
+    let clone = _.deepClone(profile);
+    _.set(clone, keySet, e.target.value);
     updateProfile(clone);
   };
   return (
     <TextField
       floatingLabel
       label={label}
-      value={profile[key]}
-      error={error}
+      value={_.get(profile, keySet, undefined)}
```

Comment:
haha, somehow I decided that it returned something else by default :stuck_out_tongue_closed_eyes: